src/model/Patterns.ts

```diff
@@ -0,0 +1,109 @@
+export type Level = "Error" | "Info" | "Warning"
+
+export type Category =
+  | "ErrorProne"
+  | "CodeStyle"
+  | "Complexity"
+  | "UnusedCode"
+  | "Security"
+  | "Compatibility"
+  | "Performance"
+  | "Documentation"
+  | "BestPractice"
+
+export function fromEslintCategoryToCategory(category?: string): Category {
+  switch (category) {
+    case "Possible Errors":
+      return "ErrorProne"
+    case "Best Practices":
+      return "BestPractice"
+    case "Strict Mode":
+      return "BestPractice"
+    case "Variables":
+      return "CodeStyle"
+    case "Node.js and CommonJS":
+      return "BestPractice"
+    case "Stylistic Issues":
+      return "CodeStyle"
+    case "ECMAScript 6":
+      return "BestPractice"
+    case "Deprecated":
+      return "Compatibility"
+    case "Removed":
+      return "Compatibility"
+    default:
+      return "CodeStyle"
+  }
+}
+
+export function fromEslintCategoryToLevel(category?: string): Level {
+  switch (category) {
+    case "Possible Errors":
+      return "Error"
+    case "Best Practices":
+      return "Info"
+    case "Strict Mode":
+      return "Error"
+    case "Variables":
+      return "Info"
+    case "Node.js and CommonJS":
+      return "Warning"
+    case "Stylistic Issues":
+      return "Info"
+    case "ECMAScript 6":
+      return "Warning"
+    case "Deprecated":
+      return "Info"
+    case "Removed":
+      return "Info"
+    default:
+      return "Info"
+  }
+}
+
+export function patternIdToCodacy(patternId: string): string {
+  return patternId.split("/").join("_")
+}
+
+export function patternIdToEslint(patternId: string): string {
+  return patternId.split("_").join("/")
```

Comment:
use regex instead

Reply:
and this is not safe ...

Reply:
I know it's not safe.. But:
1) couldn't find rules with underscore names.
2) did this for retro-compatibility with the old tool, so old patterns' settings are maintained.

Reply:
✅ Moved to regex, added test too.